scripts/generate-topology.py

```diff
@@ -19,17 +20,40 @@ def main():
     p.add_argument('-i', dest='input', required=True)
     p.add_argument('-o', dest='output', required=True)
     p.add_argument('-c', dest='count', type=int, default=1)
-    p.add_argument('--generated-seed', dest='generated_seed', action='store_true', default=False)
+    p.add_argument('--seed', dest='seed_policy', type=str, default="r", \
+        help="Set seed policy. 'r' for random seed, 'g' for generated seed, 'f' for fixed seed")
     p.add_argument('--topology', dest='topology', default=None)
     p.add_argument('--min-distance', dest='min_distance', type=int, default=0)
+    p.add_argument('--tx_rat', dest='tx_ratio', type=float, default=1.0, nargs='+', help="Transmision ratio of network. Values will be rounded up to 2 digits.")
+    p.add_argument('--rx_rat', dest='rx_ratio', type=float, default=1.0, nargs='+', help="Receive ratio of network. Values will be rounded up to 2 digits.")
     try:
         args = p.parse_args(sys.argv[1:])
     except Exception as e:
         sys.exit(f"Illegal arguments: {str(e)}")
+    
+    args.seed_policy = args.seed_policy.lower()
+    if args.seed_policy not in ['r', 'g', 'f']:
+        print(f"Seed policy must be one of 'r', 'g', 'f' ({args.seed_policy} was given)")
+        sys.exit("Invalid seed policy")
+    
+    if type(args.tx_ratio) == float:
+        args.tx_ratio = [args.tx_ratio]
+    if type(args.rx_ratio) == float:
+        args.rx_ratio = [args.rx_ratio]
```

Comment:
These arguments are already lists as they are defined with `nargs='+'`. No conversion should be needed.